cmd/up/main.go

```diff
@@ -483,22 +450,33 @@ func buildOptionsFromFlags(
 	return opts, err
 }
 
-func tokenProvider(token, tokenFile string) TokenProvider {
-	var res TokenProvider
+func setEndpointType(opts *options.Options, rawEndpointType string) {
+	switch options.EndpointType(rawEndpointType) {
+	case options.LogsEndpointType:
+		opts.EndpointType = options.LogsEndpointType
+	case options.MetricsEndpointType:
+		opts.EndpointType = options.MetricsEndpointType
+	default:
+		panic("unexpected endpoint type")
```

Comment:
if someone incorrectly types `log` instead of `logs` then this will cause the program to panic. Can we make this friendlier for the user and return an error?